src/Connection/Http1Connection.php

```diff
@@ -449,10 +453,9 @@ private function determineKeepAliveTimeout(Response $response): int
 
         $params = Http\createFieldValueComponentMap(Http\parseFieldValueComponents($response, 'keep-alive'));
 
-        return $this->priorTimeout = \min(
-            \max(0, ((int) ($params['timeout'] ?? $this->priorTimeout)) - 1),
-            self::MAX_KEEP_ALIVE_TIMEOUT
-        );
+        $timeout = (int) ($params['timeout'] ?? $this->priorTimeout);
+
+        return $this->priorTimeout = \min(\max(0, $timeout - 1), self::MAX_KEEP_ALIVE_TIMEOUT);
```

Comment:
I guess the timeout here should be the real timeout. Instead we should avoid trying non-retry-able requests if we're close to the timeout.

Reply:
Sure, but then why not just close the connection because it won't get used again anyway?

Reply:
A `GET` request coming shortly after might still use that connection, because it can be safely retried if it fails.

Reply:
Yes, I missed "non-retry-able."

Pushed an update similar to what you did then.

Reply:
In an optimal world that time would be based on the RTT to the server, it probably makes sense to make it configurable.